src/server/app/util/services/schema.server.service.js

```diff
@@ -1,13 +1,106 @@
 'use strict';
 
-var mongoose = require('mongoose'),
+const _ = require('lodash'),
+	mongoose = require('mongoose'),
 	Schema = mongoose.Schema;
 
-exports.GetterSchema = function (add) {
+/**
+ * Instructs Mongoose to use getters for the toObject and toJSON methods
+ * for an input schema definition
+ */
+const GetterSchema = function (add) {
 	var schema = new Schema(add);
 
 	schema.set('toObject', { getters: true });
 	schema.set('toJSON', { getters: true });
 
 	return schema;
-};
\ No newline at end of file
+};
+
+/**
+ * Translates the input sort array to an object with the properties specified.
+ * Defaults to descending sort on each property
+ */
+const generateSort = (sorting) => {
+
+	if(_.isArray(sorting)) {
+		var sortObj = {};
+
+		// Extract the sort instructions with defaults for DESC sort on each property
+		_.forEach(sorting, (d) => {
+			if(!_.isEmpty(d) && !_.isEmpty(d.property)) {
+				sortObj[d.property] = (d.direction === 'ASC')? 1 : -1;
+			}
+		});
+
+		return sortObj;
+	}
+
+	/*
+	 * Otherwise, if it is not an array, assume that the value passed-in is
+	 * a sorting object that the caller expected to be sent directly to the query
+	 */
+	return sorting;
+
+};
+
+/**
+ * Adds a static method 'countSearch' to the schema that performs concurrent
```

Comment:
Curious how the countSearch method gets defined when it's called pagingSearch below and I don't see any reference to countSearch?